core/src/test/java/io/forestframework/core/http/HandlerIntegrationTest.java

```diff
@@ -0,0 +1,458 @@
+        assertThat(result, containsString(handler));
+        assertThat(result, containsString("is handled by custom 400errorhandler"));
+
+        checkConsoleLogPrintingIn(result);
+    }
+
+    @Test
+    void exceptionsInPostHandlersAreHandledByFallbackErrorHandlerNotCustomHandler() throws IOException {
+        String result = sendHttpRequest("GET", "/errorInPostHandler/random");
+
+        assertThat(errorInPostHandler.traces, not(hasItem("shouldNotBeHandledByCustomErrorHandler")));
+        Assertions.assertEquals(
+                Collections.synchronizedList(Arrays.asList(
+                        "handler",
+                        "errorInPostHandler"
+                )), errorInPostHandler.traces);
+    }
+
+    @Test
+    void uncaught404ErrorIsHandledByFallbackErrorHandler() throws IOException {
+        String result = sendHttpRequest("GET", "/uncaught404/inexistence");
+
+        assertThat(uncaught404Error.traces, not(hasItem("uncaught404")));
+        assertThat(uncaught404Error.traces, not(hasItem("custom500ErrorHandler")));
+        Assertions.assertEquals(HttpStatusCode.NOT_FOUND.name(), result);
+
+        checkConsoleLogPrintingIn(result);
+    }
+
+    @Test
+    void httpExceptionCanBeThrownFromPreHandlersToPreventPropagation() throws IOException {
+        String result = sendHttpRequest("GET", "/prehandlerError/random");
+
+        Assertions.assertEquals(
+                Arrays.asList(
+                        "preHandlerThrowsException",
+                        "shouldBeHandledByCustomErrorHandler",
+                        "shouldContinueInPostHandler"
+                ), prehandlerErrorPreventPropagation.traces);
+        assertThat(result, containsString(HttpStatusCode.BAD_REQUEST.name()));
+        assertThat(result, containsString("should be handled by custom error handler"));
+
+        checkConsoleLogPrintingIn(result);
+    }
+
+    @Test
+    void shouldThrowExceptionsWhenPreHandlersNotReturnBoolean() throws IOException {
+        String result = sendHttpRequest("GET", "/preHandlerNotReturnBoolean/random");
+
+        Assertions.assertEquals(
+                Arrays.asList(
+                        "prehandlerReturnsString",
+                        "customErrorHandler",
+                        "postHandler"
+                ), throwExceptionsWhenPreHandlersNotReturnBoolean.traces);
+        assertThat(result, containsString("java.lang.ClassCastException"));
+        assertThat(result, containsString("class java.lang.String cannot be cast to class java.lang.Boolean"));
+    }
+
+    private String sendHttpRequest(String method, String path) throws IOException {
+        String uri = "http://localhost:" + port + path;
+
+        HttpUriRequest request = RequestBuilder.create(method).setUri(uri).build();
+        CloseableHttpResponse response = client.execute(request);
+        HttpEntity entity = response.getEntity();
```

Comment:
Why not check status code? We should check if the status code is expected, esp. in the case of 4xx/5xx.

Reply:
'kay, will add status code checking.